### DISCUSSION TITLE
Eventhandler to BackstageAppButton/Home button

### DISCUSSION THREAD

**Discussion by kmurari103** [iTwin/itwinjs-core#3051] [Q&A] [answered]
Hi there,
I am running into an issue. 
I have created a modeless dialog which is displayed when a toolbar button is clicked, I can close this dialog using '**X'** present at dialog's corner.
I also want to close it  when someone click on home button. **The reason for this is**- I open backstage then User Profile is opened but User Profile window also shows the dialog that i opened earlier from toolbar button which should be closed ideally.
Is there a way to close that dialog when home button is clicked  in addition to closing it from dialog's **X** itself.

**Comment by bsteinbk** [comment, marked as answer]
@kmkashyap - we don't auto close any popups when the backstage is opened or closed. One possible solution is to add a listener to be notified when the backstage is opened or closed and then call API to close dialogs. Below is some sample code to try.

```
Backstage.onBackstageEvent.addListener(bsCallback);

function bsCallback () {
  // to close just modeless dialogs
  ModelessDialogManager.closeAll();

  // to close all popups including modeless dialogs
  ConfigurableUiManager.closeUi()
}

```

**Reply by kmurari103** [reply]
@bsteinbk Thanks for the reply but `Backstage.onBackstageEvent.` is deprecated as per [this link](https://github.com/iTwin/itwinjs-core/blob/release/2.19.x/ui/framework/src/ui-framework/backstage/Backstage.tsx#:~:text=/**%20Backstage%20React%20component,*/) and suggestion is to use `BackstageComposer` but i don't see any event related props there. 
The reason i'm looking to close any dialog/popup when home button clicked can be seen in the attached snapshot. It does not look good to see those dialogs. 

![image](https://user-images.githubusercontent.com/27068008/150078909-24176b32-5784-4eb5-af35-fbd2bdc51ae0.png)

Is there any workaround for this?

**Reply by bsteinbk** [reply]
@kmkashyap - marking the Backstage as deprecated was an error. I will soon be opening a PR to remove the deprecation. The Backstage component is still used, The BackstageComposer composes the items to show in the Backstage. It should be safe to use the method I suggested.
